### PR TITLE
Work with jdk9

### LINKED ISSUE #1172: Ensure compile with JDK 9
We should check if Travis supports it already

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #1172 

It actually didn't compile with JDK 9 because of an outdated plugin...

Took the opportunity to update a few other plugins as well, including the JUnit plugin!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ language: java
 install: mvn install -DskipTests=true
 script: mvn test
 jdk:
-  - oraclejdk8
+  - oraclejdk9
```
Comment (ftomassetti) [inline .travis.yml]: I would not replace it but add it: Travis should be able to run the tests using both the JDKs so we verify it stays compatible with JDK8 and JDK9 at the same time

File: javaparser-core/pom.xml [hunk 2]
```diff
@@ -79,7 +79,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-enforcer-plugin</artifactId>
-                <version>1.4.1</version>
+                <version>3.0.0-M1</version>
```
Comment (ftomassetti) [inline javaparser-core/pom.xml:82]: This M1 should scare us?

File: .travis.yml (hunk 1 above)
Comment (matozoid) [reply .travis.yml]: How would I do that?

File: javaparser-core/pom.xml (hunk 2 above)
Comment (matozoid) [reply javaparser-core/pom.xml:82]: Not as much as 1.4.2-jenkins1 ;-) After some hunting for release notes I found this: http://maven.40175.n5.nabble.com/ANN-Apache-Maven-Enforcer-Plugin-3-0-0-M1-Released-td5911744.html

File: .travis.yml (hunk 1 above)
Comment (ftomassetti) [reply .travis.yml]: I think you can just add the second line without removing the first one

Comment (matozoid) [reply .travis.yml]: It worked :-O

File: javaparser-core/pom.xml (hunk 2 above)
Comment (matozoid) [reply javaparser-core/pom.xml:82]: Anyway, the alternatives are worse in my opinion: either take out the JDK 8 API enforcement for core, or don't compile with JDK9 at all.

**Comment by matozoid** [comment]
Assuming okay, @ftomassetti please comment if you disagree so I can do more.

**Comment by ftomassetti** [comment]
It seems perfect to me

**Pull request merged**